### PR TITLE
Add samples for MLContext.Model (root) and add links to documentation.

### PR DESCRIPTION
**Pull request by codemzs**
Towards #1209

### CODE REVIEW

File: src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs [hunk 1]
```diff
@@ -281,6 +302,13 @@ public ITransformer LoadWithDataLoader(string filePath, out IDataLoader<IMultiSt
         /// <typeparamref name="TDst"/>.</param>
         /// <param name="inputSchemaDefinition">Additional settings of the input schema.</param>
         /// <param name="outputSchemaDefinition">Additional settings of the output schema.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[Load](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/TensorFlow/TextClassification.cs)]
```
Comment (shmoradims) [inline src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs]: >TextClassification [](start = 103, length = 18)

this looks too complicated. let's use a simpler pipeline with bare minimum (no metrics, no testing, no textloader, just Fit and CreatePredictEngine). You can take one of our simpler examples and cut the unnecessary parts.

File: src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs [hunk 2]
```diff
@@ -187,6 +201,13 @@ public ITransformer Load(Stream stream, out DataViewSchema inputSchema)
         /// <param name="inputSchema">Will contain the input schema for the model. If the model was saved without
         /// any description of the input, there will be no input schema. In this case this can be <see langword="null"/>.</param>
         /// <returns>The loaded model.</returns>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[Load](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/CustomMappingSaveAndLoad.cs)]
```
Comment (shmoradims) [inline src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs]: >CustomMappingSaveAndLoad [](start = 103, length = 24)

I think majority of the Load()/Save() operations don't require mlContext.ComponentCatalog.RegisterAssembly that CustomMapping requires, which makes this sample the exception not the rule. If that's assumption is correct, let's stick to a sample that doesn't scare users with using  RegisterAssembly, CustomMappingFactoryAttribute, and Action<>. You can use the same approach I suggested for CreatePredictionEngine sample below (simple data, simple pipeline) and just simple Save() then Load().

CustomMappingSaveAndLoad is best suited for CustomMapping API which we already use it there.

File: src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs [hunk 3]
```diff
@@ -135,6 +142,13 @@ private void SaveInputSchema(DataViewSchema inputSchema, RepositoryWriter rep)
         /// <param name="inputSchema">Will contain the input schema for the model. If the model was saved without
         /// any description of the input, there will be no input schema. In this case this can be <see langword="null"/>.</param>
         /// <returns>The loaded model.</returns>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[Load](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectChangePointBySsa.cs]
```
Comment (shmoradims) [inline src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs]: >DetectChangePointBySsa [](start = 114, length = 22)

Same here. please use/create a sample with the bare minimum required to show the API usage.

File: src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs (hunk 2 above)
Comment (sfilipi) [reply src/Microsoft.ML.Data/Model/ModelOperationsCatalog.cs]: Agree for this one, CustomMapping might distract them. Maybe a simpler pipeline. 

---
In reply to: [275778414](https://github.com/dotnet/machinelearning/pull/3352#discussion_r275778414) [](ancestors = 275778414)

**Pull request merged**